closed/src/java.base/share/classes/sun/security/ec/NativeECDHKeyAgreement.java

```diff
@@ -136,7 +142,12 @@ private void init(Key key)
                 this.initializeJavaImplementation(key);
                 return;
             }
-            this.privateKeyOps = opsOpt.get();
+
+            if (ECUtil.isBrainpoolP512r1(ecPrivateKey.getParams())) {
+                this.privateKeyOps = null;
+            } else {
+                this.privateKeyOps = opsOpt.get();
+            }
```

Comment:
With the suggestion above, we already have the value in a local:
```java
            this.privateKeyOps = ops;
```

Reply:
Updated as suggested.

Reply:
I don't see that update in 2705bd931da889b65b7f51149b4a321199272f1c.

Reply:
Perhaps I am not sure what is being suggested here. I update this code from:
```
} else {
    this.privateKeyOps = opsOpt.get()
}
```
to
```
} else {
    this.privateKeyOps = ops;
}
```
Perhaps there is another update being requested here?

Reply:
Replaced this code with simply `this.privateKeyOps = ops;`